File: 0001_TwoSum.py
class Solution:
    def twoSum(self, nums: list[int], target: int) -> list[int]:
        d = {}
        for i, elem in enumerate(nums):
            elem2 = target - elem
            if elem2 in d:
                return [i, d[elem2]]
            d[elem] = i

File: 0001_TwoSum.go
func twoSum(A []int, target int) []int {
    m1 := make(map[int]int)
    for i, num := range A {
        value, exists := m1[target - num]
        if exists {
            return []int{value, i}
        } else {
            m1[num] = i
        }
    }
    return []int{}
}